### ISSUE TITLE
Can't use GitHub panel for a private repo inside my company - "Query errors reported: could not resolve 'name'"

### ISSUE THREAD

**Issue by jpvalery** [atom/github#2093]
### Description

I'm facing an annoying (but tolerable) issue with the Github panel

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:**

[What did you expect to happen?]

**Actual behavior:**

[What actually happened instead?]

**Reproduces how often:**

[What percentage of the time does this happen?]

### Versions

You can get this information from copy and pasting the output of `atom --version` and `apm --version` from the command line. Also, please include the OS and what version of the OS you're running.

### Additional Information

Please include any additional information, configuration or data that might be necessary to reproduce the issue. Screenshots, if appropriate, are helpful as well.

If your issue involves a repository, and that repository is public, please include a link to it.

**Comment by rsese** [comment]
Thanks for the report!  To clarify, are you only having issues with your company's private repositories and otherwise have no issues with a repository owned by your user account for example?

**Comment by jpvalery** [comment]
Exactly! Hence my surprise :)

Tried talking with the person in charge of our repos and he had no idea as to why it behaves this way.

**Comment by rsese** [comment]
Hmm, though the error message is different, does this apply to your organization?

https://github.com/atom/github/issues/1010#issuecomment-422882361

**Comment by jpvalery** [comment]
IDK. I shared this with the person responsible for our GitHub and will report back if/when I hear back.

**Comment by RinardNick** [comment]
This is a problem for me as well, "Query errors reported" Could not resolve to a Repository with the name '******'

**Comment by jquinn-antaeus** [comment]
I was having a similar issue that was resolved by the repo admin re-sending the invitation to the repo.  Once I accepted the invitation, I could see the repository in search results and the repo appeared in the "repositories" sidebar of my github.com main page.  Entering the key into Atom connected to GitHub with no further errors.

**Comment by rsese** [comment]
@jquinn-antaeus were you also seeing the "Query errors reported" Could not resolve to a Repository with the name" error message?

For folks seeing that specific error, do you happen to be using repositories hosted on a GitHub Enterprise instance?

**Comment by jquinn-antaeus** [comment]
> @jquinn-antaeus were you also seeing the "Query errors reported" Could not resolve to a Repository with the name" error message?

Yes.  The search for that error led me to this thread.

**Comment by rogerkk** [comment]
@rsese I see this error, but I'm not using Github Enterprise.

**Comment by Si-Thu-Thein** [comment]
I also have this error with one of my repos in my organization

**Comment by rsese** [comment]
Is there anything different about the repository that you can think of versus the rest of the repositories in your organization @Si-Thu-Thein?  Do you have the same permissions on this particular repository?

**Comment by olbrich** [comment]
I am seeing errors like this when trying to look at an 'internal' repository in our corporate organization which is using Github enterprise cloud.  I think we also have SSO enabled as I am asked to do that from time to time when using the Github UI.

**Comment by Si-Thu-Thein** [comment]
@rsese I don't think this repo has any special permissions at all, except that it's a private one. But another private repo is working fine...

**Comment by HadrienPatte** [comment]
If it helps, I have the `Query errors reported: could not resolve 'name'` for a private repository in an organization that requires 2FA and I don't have it on private repositories with no 2FA required.

**Comment by danbrannon** [comment]
I don't have 2FA enabled.
I'm getting this error on an organization's private repo .wiki as well.

**Comment by FXschwartz** [comment]
I am getting this error on my organization's private repo as well! `Query errors reported: could not resolve to a repository with the name 'name'`

**Comment by kkhanh** [comment]
Same on my organization

**Comment by Ghormdog** [comment]
You need your GitHub Admin to go to "Third-Party Access" to approve the access request.  Its how we solved this issue on our side.

![image](https://user-images.githubusercontent.com/20526419/70169795-84150980-1699-11ea-9c6f-0737ff830691.png)

**Comment by FXschwartz** [comment]
@Ghormdog I am a GitHub Admin for our organization and there is not even an option under Third-party access for `Atom GitHub package`. How did you request access through Atom?

**Comment by jamesr-3rdmill** [comment]
> 
> 
> @Ghormdog I am a GitHub Admin for our organization and there is not even an option under Third-party access for `Atom GitHub package`. How did you request access through Atom?
> 
> ![image](https://user-images.githubusercontent.com/16527593/70241830-567d9e00-1735-11ea-9fb5-405425b3c76c.png)

Could this be helpful?  I was having similiar issues and followed this:

https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/requesting-organization-approval-for-oauth-apps

**Comment by FXschwartz** [comment]
@jamesr-3rdmill That did it!

Thank you so much for your help! All is working now.

**Comment by Ghormdog** [comment]
Thanks all - appreciate it!

**Comment by ivo-ivanov** [comment]
@Ghormdog thank you, this solved my issue!

**Comment by billhofmann-fade** [comment]
Nope, not working in 1.51.0.  Query errors reported, Could not resolve to a Repository with name '<org>/<repo>'. Console reports uncaught throw:
Uncaught (in promise) Error: GraphQL API endpoint at https://api.github.com/graphql returned an error for query GetMentionableUsers.
    at fetchQuery (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:14)
    at UserStore.loadMentionableUsers (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:14)
    at async Promise.all (index 0)
    at UserStore.loadUsers (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:14)

**Comment by ErrorBot1122** [comment]
has anyone found an answer to this issue?

**Comment by bstivers** [comment]
The above third-party access worked for me with some slight menu tweaks. Make sure it's approved on the organization side as well as your own profile side. For more info, see here:

https://docs.github.com/en/organizations/restricting-access-to-your-organizations-data/about-oauth-app-access-restrictions
